src/components/dropdownSearch/DropdownSearchReducers.tsx

```diff
@@ -111,11 +154,17 @@ export const dropdownSearchReducer = (state: IDropdownSearchState = dropdownSear
         return {
           ...state,
           id: action.payload.id,
-          selectedOption: state.activeOption,
           isOpened: false,
+          selectedOptions: state.selectedOptions.push(state.activeOption),
```

Comment:
It's not about the side effects, it's about the state being read-only and immutable. You should be able to go backward and be able to see what the state was at any moment.